app/auth/routes.py

```diff
@@ -1,21 +1,32 @@
 from flask.views import MethodView
 from flask import Blueprint, jsonify, request
 from app.models import User
+from dbconnection import dbConnection
 
 auth = Blueprint("auth", __name__)
+db_connection = dbConnection()
 
 class RegisterUser(MethodView):
     """ Route class to register a user """
     def post(self):
         """ Register a user """
                 
         post_data = request.get_json()
+
         username = post_data.get('username')
         email = post_data.get('email')
         password = post_data.get('password')
         isAdmin = post_data.get('isAdmin')
+
+        if not username:
```

Comment:
Do better validation. This is just testing for a condition where the specific data is not passed, but take a scenario where I pass an empty string or white space or symbols. You should take those and which ever other edge cases into consideration. Also, it would be better to create a utilises class that has all those validation methods and they are just called as opposed to each endpoint having to validate the data on it's own